From 6685e7a99f3a3a1a144d304762e1486cf47c9a48 Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Mon, 2 Jun 2014 19:04:31 +0300
Subject: [PATCH 1730/1825] msys_ac3: updated SatR register mapping in DFX
 server

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit bcd8206b88847c5a1ab01f301baa9623170e3f4d

	Alley-Cat3 and Bobcat2 Differ in some register mapping
	- Updated Sample at reset register mapping that are used by 'SatR' API's
	- Added differentiation for Boot mode, to indicate which mapping to use

Change-Id: Ib65d4ff97fb3d6d2efaa4a986b7e58a19a8b48ae
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/8383
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/8582
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../mach-msys/msys_family/ctrlEnv/mvCtrlEnvLib.c   |   21 ++++++++++++++-----
 .../mach-msys/msys_family/ctrlEnv/mvCtrlEnvRegs.h  |   10 ++++++--
 2 files changed, 22 insertions(+), 9 deletions(-)

diff --git a/arch/arm/mach-msys/msys_family/ctrlEnv/mvCtrlEnvLib.c b/arch/arm/mach-msys/msys_family/ctrlEnv/mvCtrlEnvLib.c
index ebe4c85..24e2b8d 100644
--- a/arch/arm/mach-msys/msys_family/ctrlEnv/mvCtrlEnvLib.c
+++ b/arch/arm/mach-msys/msys_family/ctrlEnv/mvCtrlEnvLib.c
@@ -1072,9 +1072,12 @@ MV_U32 ctrlSizeRegRoundUp(MV_U32 size, MV_U32 alignment)
 *******************************************************************************/
 MV_BOOL mvCtrlIsBootFromNOR(MV_VOID)
 {
-	MV_U32 satr;
+	MV_U32 satr = MV_DFX_REG_READ(DFX_DEVICE_SAR_REG(0));
 
-	satr = MSAR_BOOT_MODE(MV_DFX_REG_READ(DFX_DEVICE_SAR_REG(0)), 0);
+	if (mvCtrlDevFamilyIdGet(0) == MV_BOBCAT2_DEV_ID)
+		satr = MSAR_BC2_BOOT_MODE(satr, 0);
+	else
+		satr = MSAR_AC3_BOOT_MODE(satr, 0);
 
 	if (satr == SAR1_BOOT_FROM_NOR)
 		return MV_TRUE;
@@ -1100,9 +1103,12 @@ MV_BOOL mvCtrlIsBootFromNOR(MV_VOID)
 *******************************************************************************/
 MV_BOOL mvCtrlIsBootFromSPI(MV_VOID)
 {
-	MV_U32 satr;
+	MV_U32 satr = MV_DFX_REG_READ(DFX_DEVICE_SAR_REG(0));
 
-	satr = MSAR_BOOT_MODE(MV_DFX_REG_READ(DFX_DEVICE_SAR_REG(0)), 0);
+	if (mvCtrlDevFamilyIdGet(0) == MV_BOBCAT2_DEV_ID)
+		satr = MSAR_BC2_BOOT_MODE(satr, 0);
+	else
+		satr = MSAR_AC3_BOOT_MODE(satr, 0);
 
 	if (satr == SAR1_BOOT_FROM_SPI)
 		return MV_TRUE;
@@ -1128,9 +1134,12 @@ MV_BOOL mvCtrlIsBootFromSPI(MV_VOID)
 *******************************************************************************/
 MV_BOOL mvCtrlIsBootFromNAND(MV_VOID)
 {
-	MV_U32 satr;
+	MV_U32 satr = MV_DFX_REG_READ(DFX_DEVICE_SAR_REG(0));
 
-	satr = MSAR_BOOT_MODE(MV_DFX_REG_READ(DFX_DEVICE_SAR_REG(0)), 0);
+	if (mvCtrlDevFamilyIdGet(0) == MV_BOBCAT2_DEV_ID)
+		satr = MSAR_BC2_BOOT_MODE(satr, 0);
+	else
+		satr = MSAR_AC3_BOOT_MODE(satr, 0);
 
 	if (satr == SAR1_BOOT_FROM_NAND)
 		return MV_TRUE;
diff --git a/arch/arm/mach-msys/msys_family/ctrlEnv/mvCtrlEnvRegs.h b/arch/arm/mach-msys/msys_family/ctrlEnv/mvCtrlEnvRegs.h
index 627a55a..0f230d4 100644
--- a/arch/arm/mach-msys/msys_family/ctrlEnv/mvCtrlEnvRegs.h
+++ b/arch/arm/mach-msys/msys_family/ctrlEnv/mvCtrlEnvRegs.h
@@ -161,12 +161,16 @@ extern "C" {
 /* Sample at Reset */
 #define DFX_DEVICE_SAR_REG(x)			(0xf8200 +(x*4))
 
-#define MSAR_CORE_CLK(sar1, sar2)		(((sar2) >> 21) & 0x7)		/* PLL 0 config */
+#define MSAR_CORE_CLK(sar1, sar2)	(((sar2) >> 21) & 0x7)		/* PLL 0 config */
 #define MSAR_CPU_DDR_CLK(sar1, sar2)	(((sar2) >> 18) & 0x7)		/* PLL 1 config */
-#define MSAR_TM_CLK(sar1, sar2)			(((sar2) >> 15) & 0x7)		/* PLL 2 config */
-#define MSAR_BOOT_MODE(sar1, sar2)		(((sar1) >> 13) & 0x7)		/* boot from */
 #define MSAR_DEVICE_MODE(sar1, sar2)	(((sar1) >>  0) & 0xFF)		/* DEVICE ID field */
 
+#define MSAR_BC2_BOOT_MODE(sar1, sar2)	(((sar1) >> 13) & 0x7)		/* boot from */
+#define MSAR_BC2_TM_CLK(sar1, sar2)	(((sar2) >> 15) & 0x7)		/* PLL 2 config */
+
+#define MSAR_AC3_BOOT_MODE(sar1, sar2)	(((sar1) >> 11) & 0x7)		/* boot from */
+#define MSAR_AC3_TM_CLK(sar1, sar2)	(((sar2) >> 17) & 0x1)		/* PLL 2 config */
+
 #define SAR1_BOOT_FROM_NOR			0
 #define SAR1_BOOT_FROM_NAND			1
 #define SAR1_BOOT_FROM_UART			2
-- 
1.7.5.4

